### ISSUE TITLE
Error installing DOSE

### ISSUE THREAD

**Issue by bimbam23** [GuangchuangYu/clusterProfiler#125]
Hi,

I'm having problems installing clusterProfiler. Here is the error message:

```
install_github("GuangchuangYu/clusterProfiler")
Downloading GitHub repo GuangchuangYu/clusterProfiler@master
from URL https://api.github.com/repos/GuangchuangYu/clusterProfiler/zipball/master
Installing clusterProfiler
trying URL 'https://bioconductor.org/packages/3.6/bioc/src/contrib/DOSE_3.4.0.tar.gz'
Content type 'application/x-gzip' length 7573499 bytes (7.2 MB)
==================================================
downloaded 7.2 MB

Installing DOSE
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet  \
  CMD INSTALL '/tmp/RtmpmfISav/devtools166b312c4353/DOSE'  \
  --library='/home/bickj/R/x86_64-pc-linux-gnu-library/3.4' --install-tests 

* installing *source* package ‘DOSE’ ...
** R
** data
** inst
** tests
** byte-compile and prepare package for lazy loading
Warning: S3 methods ‘$.igraph’, ‘$.igraph.es’, ‘$.igraph.vs’, ‘$<-.igraph’, ‘$<-.igraph.es’, ‘$<-.igraph.vs’, ‘*.igraph’, ‘+.igraph’, ‘-.igraph’, ‘[.communities’, ‘[.igraph’, ‘[.igraph.es’, ‘[.igraph.vs’, ‘[<-.igraph’, ‘[<-.igraph.es’, ‘[<-.igraph.vs’, ‘[[.communities’, ‘[[.igraph’, ‘[[.igraph.es’, ‘[[.igraph.vs’, ‘[[<-.igraph.es’, ‘[[<-.igraph.vs’, ‘as.dendrogram.communities’, ‘as.hclust.communities’, ‘as.igraph.igraphHRG’, ‘as_ids.igraph.es’, ‘as_ids.igraph.vs’, ‘as_phylo.communities’, ‘c.igraph.es’, ‘c.igraph.vs’, ‘cohesion.cohesiveBlocks’, ‘cohesion.igraph’, ‘compare.communities’, ‘compare.default’, ‘compare.membership’, ‘difference.igraph’, ‘difference.igraph.es’, ‘difference.igraph.vs’, ‘graph_id.igraph’, ‘graph_id.igraph.es’, ‘graph_id.igraph.vs’, ‘groups.communities’, ‘groups.default’, ‘intersection.igraph’, ‘in [... truncated]
Error in library.dynam(lib, package, package.lib) : 
  shared object ‘igraph.so’ not found
ERROR: lazy loading failed for package ‘DOSE’
* removing ‘/home/bickj/R/x86_64-pc-linux-gnu-library/3.4/DOSE’
Installation failed: Command failed (1)
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet  \
  CMD INSTALL  \
  '/tmp/RtmpmfISav/devtools166b5d695792/GuangchuangYu-clusterProfiler-19c4dea'  \
  --library='/home/bickj/R/x86_64-pc-linux-gnu-library/3.4' --install-tests 

ERROR: dependencies ‘DOSE’, ‘enrichplot’ are not available for package ‘clusterProfiler’
* removing ‘/home/bickj/R/x86_64-pc-linux-gnu-library/3.4/clusterProfiler’
Installation failed: Command failed (1)
```

then I manage to install DOSE alone but I'm getting this:

```
install_github("GuangchuangYu/clusterProfiler")
Downloading GitHub repo GuangchuangYu/clusterProfiler@master
from URL https://api.github.com/repos/GuangchuangYu/clusterProfiler/zipball/master
Installing clusterProfiler
'/usr/lib/R/bin/R' --no-site-file --no-environ --no-save --no-restore --quiet CMD INSTALL  \
  '/tmp/RtmpJE2vR4/devtools58d43e421dd8/GuangchuangYu-clusterProfiler-19c4dea'  \
  --library='/home/bickj/R/x86_64-pc-linux-gnu-library/3.4' --install-tests 

ERROR: dependency ‘enrichplot’ is not available for package ‘clusterProfiler’
* removing ‘/home/bickj/R/x86_64-pc-linux-gnu-library/3.4/clusterProfiler’
Installation failed: Command failed (1)
```

Cheers Jochen

**Comment by bimbam23** [comment]
``

**Comment by GuangchuangYu** [comment]
To install github version of clusterProfiler, you need to use the following commands:

```r
install_github(c("guangchuangyu/DOSE",
                         "guangchuangyu/enrichplot",
                         "guangchuangyu/clusterProfiler")
```

Currently, `enrichplot` is only available for R 3.5, see <https://github.com/Bioconductor/Contributions/issues/567> and <https://github.com/GuangchuangYu/enrichplot/issues/2>.

**Comment by GuangchuangYu** [comment]
see <https://github.com/GuangchuangYu/enrichplot/issues/3>.
